src/mmrelay/db_utils.py

```
@@ -37,27 +46,103 @@ def __init__(self, table: str) -> None:
+    PROTO_NODE_NAME_LONG: _NAME_FIELD_BY_TABLE[NAMES_TABLE_LONGNAMES],
+    PROTO_NODE_NAME_SHORT: _NAME_FIELD_BY_TABLE[NAMES_TABLE_SHORTNAMES],
+}
+_LONGNAME_DB_FIELD = _NAME_FIELD_BY_TABLE[NAMES_TABLE_LONGNAMES]
+_SHORTNAME_DB_FIELD = _NAME_FIELD_BY_TABLE[NAMES_TABLE_SHORTNAMES]
+
+# SQL templates use constant table/column names (not user input) with parameterized values.
+# Ruff S608 warnings are false positives - NAMES_TABLE_* and NAMES_FIELD_* are constants.
 _SELECT_STALE_IDS_SQL_BY_TABLE = {
-    "longnames": "SELECT meshtastic_id FROM longnames",
-    "shortnames": "SELECT meshtastic_id FROM shortnames",
+    NAMES_TABLE_LONGNAMES: f"SELECT meshtastic_id FROM {NAMES_TABLE_LONGNAMES}",
```

Ruff (0.15.6)

[error] Possible SQL injection vector through string-based query construction

(S608)